CRM/Report/Form/Mailing/Summary.php

```diff
@@ -297,23 +295,8 @@ public function __construct() {
         ),
       ),
     );
-    $config = CRM_Core_Config::singleton();
-    $this->campaignEnabled = in_array("CiviCampaign", $config->enableComponents);
-    if ($this->campaignEnabled) {
-      $this->_columns['civicrm_campaign'] = array(
-        'dao' => 'CRM_Campaign_DAO_Campaign',
-        'fields' => array(
-          'title' => array(
-            'title' => ts('Campaign Name'),
-          ),
-        ),
-        'filters' => array(
-          'title' => array(
-            'type' => CRM_Utils_Type::T_STRING,
-          ),
-        ),
-      );
-    }
+    // If we have campaigns enabled, add those elements to both the fields, filters.
+    $this->addCampaignFields('civicrm_membership');
     parent::__construct();
```

Comment:
@seamuslee001 
Shouldn't this be _civicrm_mailing_ intead of _civicrm_membership_?